src/Mod/Material/App/Materials.cpp

```diff
@@ -0,0 +1,914 @@
+/***************************************************************************
+ *   Copyright (c) 2023 David Carter <dcarter@david.carter.ca>             *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+#ifndef _PreComp_
+#endif
+
+#include <QMetaType>
+#include <QUuid>
+
+#include <App/Application.h>
+#include <Gui/MetaTypes.h>
+
+#include "MaterialManager.h"
+#include "Materials.h"
+#include "ModelManager.h"
+
+
+using namespace Materials;
+
+/* TRANSLATOR Material::Materials */
+
+TYPESYSTEM_SOURCE(Materials::MaterialProperty, Materials::ModelProperty)
+
+MaterialProperty::MaterialProperty()
+{
+    _valuePtr = new MaterialValue(MaterialValue::None);
```

Comment:
Possible memory leak! _valuePtr is a raw pointer but the destructor doesn't destroy it.
As said above consider using a std::shared_ptr or std::unique_ptr